api/v1alpha1/bundle_types.go

```diff
@@ -0,0 +1,80 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type BundleSourceType string
+
+// +union
+type BundleSource struct {
+	// +unionDiscriminator
+	// +optional
+	Type BundleSourceType
+
+	// +optional
+	Ref string
```

Comment:
At least some of the strawmen had this as a list of `refs` (for e.g. a set of kustomize layers, or a helm chart + a config, etc)

If this is a single string instead of an array, is there another idea for addressing "bundles" that require multiple inputs like that?

Reply:
IIRC the rationale earlier today was to start with something on a smaller scale and scale up to an array of references once we get a feel for how these APIs will interact and be consumed to get a couple of prototypes out the door quicker.